lang/c++/impl/Compiler.cc

```diff
@@ -286,8 +286,9 @@ static Field makeField(const Entity &e, SymbolTable &st, const string &ns) {
     const Object &m = e.objectValue();
     string n = getStringField(e, m, "name");
     vector<string> aliases;
-    if (containsField(m, "aliases")) {
-        for (const auto &alias : getArrayField(e, m, "aliases")) {
+    string aliasesName = "aliases";
+    if (containsField(m, aliasesName)) {
+        for (const auto &alias : getArrayField(e, m, aliasesName)) {
```

Comment:
I'll admit this is kind of a weird change... the compiler appears to think that the lifetime of the name passed to `getArrayField` may be linked to the lifetime of the returned reference. It doesn't appear to be from my investigation, but this happens to slightly improve performance wherever we use `containsField` + `getArrayField` together anyways, so seems a minor cost to pay.

```text
/home/gerrit/Desktop/avro/lang/c++/impl/Compiler.cc: In function ‘avro::Field avro::makeField(const json::Entity&, SymbolTable&, const std::string&)’:
/home/gerrit/Desktop/avro/lang/c++/impl/Compiler.cc:290:63: error: possibly dangling reference to a temporary [-Werror=dangling-reference]
  290 |         for (const auto &alias : getArrayField(e, m, "aliases")) {
      |                                                               ^
/home/gerrit/Desktop/avro/lang/c++/impl/Compiler.cc:290:47: note: the temporary was destroyed at the end of the full expression ‘avro::getArrayField((* & e), (* & m), std::__cxx11::basic_string<char>(((const char*)"aliases"), std::allocator<char>()))’
  290 |         for (const auto &alias : getArrayField(e, m, "aliases")) {
      |                                  ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
```